pipewire/qubes-pw-module.c

```diff
@@ -835,6 +910,67 @@ static void vchan_ready(struct spa_source *source)
     }
 }
 
+static int vchan_reconnect_cb(struct spa_loop *loop,
+                                bool async,
+                                uint32_t seq,
+                                const void *data,
+                                size_t size,
+                                void *user_data)
+{
+    struct qubes_stream *stream = user_data;
+    stop_watching_vchan(stream);
+    return 0;
+}
+
+static void qdb_cb(struct spa_source *source)
+{
+    struct impl *impl = source->data;
+    int domid;
+
+    pw_log_debug("Received event from QubesDB");
+
+    domid = get_domid_from_qdb(impl->qdb);
+
+    if (domid != impl->domid) {
+        if (domid >= 0)
+            pw_log_info("Setting new peer domain ID %d", domid);
+        else if (domid == -ENOENT) // not an error, AudioVM unset
+            pw_log_info("AudioVM unset, disconnecting from %d", impl->domid);
+        else {
+            errno = -domid;
+            pw_log_error("Cannot obtain new peer domain ID: %m");
+        }
+        impl->domid = domid;
+        for (int i = 0; i < 2; ++i) {
+            if (impl->stream[i].source.fd != -1)
+                pw_log_error("Closing %d", impl->stream[i].source.fd);
```

Comment:
I don't think it's an error, more like info?. And also, why only log message is under this "if", is that intentional?

Reply:
Because this message is not pertinent if there is no fd, we don't close -1 as this is the value saying there is no fd.